### PR TITLE
Add ability to serialize an instance of TimeZoneEngine.

### PR DESCRIPTION
**Pull request by PhotoKevin**
Mark classes as serializable and add a unit test for same.

### CODE REVIEW

File: core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineSerializationTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.ZoneId;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+*/
```
Comment (RomanIakovlev) [inline core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineSerializationTest.java:39]: Please clean up commented out code.

**Review by RomanIakovlev — CHANGES_REQUESTED** [review]
Looks good! I've left one minor request for a change. Maybe you could also write a couple of lines about this feature in the README?

File: core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineSerializationTest.java (hunk 1 above)
Comment (PhotoKevin) [reply core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineSerializationTest.java]: Oops. Commented out imports are removed.
Added some text about serialization to Readme.
Took the liberty of adding my app to the list of users.

File: core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineSerializationTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+public class TimeZoneEngineSerializationTest {
+    private static TimeZoneEngine engine = TimeZoneEngine.initialize(0, 5, 0, 5, false);
+
+    @Test
+    public void testSerialzation() {
+        File f = new File ("./Engine.cache");
+        try {
+            serializeTimeZoneEngine (f, engine);
+            TimeZoneEngine engine2 = deserializeTimeZoneEngine (f);
+            f.delete ();
```
Comment (RomanIakovlev) [inline core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineSerializationTest.java:34]: How about we add some smoke test here after deserialization, by trying to query for some time zone that's expected to be found in the selected area? This is just to make sure deserialization has restored the state of the classes correctly. E.g. something similar to this: https://github.com/RomanIakovlev/timeshape/blob/1861cd3fd83432c80b29eeb4856718b68847228a/core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineTest.java#L23

**Review by RomanIakovlev — CHANGES_REQUESTED** [review]
Thanks, that looks good! After taking another look at the unit test you have added, it occurred to me we can make it a bit more thorough (see inline comment).

File: core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineSerializationTest.java (hunk 2 above)
Comment (PhotoKevin) [reply core/src/test/java/net/iakovlev/timeshape/TimeZoneEngineSerializationTest.java]: I've changed the loaded area to match the other test and added a check.
```
assertEquals(engine.query(52.52, 13.40), engine2.query(52.52, 13.40));
```

**Review by RomanIakovlev — APPROVED** [review]
Looks good, thanks for your contribution!

**Comment by RomanIakovlev** [comment]
Fixes https://github.com/RomanIakovlev/timeshape/issues/140

**Comment by PhotoKevin** [comment]
Thank you for this package.

**Pull request merged**